controllers/metrics.go

```diff
@@ -151,6 +158,15 @@ func collectMetrics(ch chan<- prometheus.Metric, cluster *fdbv1beta2.FoundationD
 		addGauge(descProcessGroupMarkedRemoval, float64(removals[pclass]), string(pclass))
 		addGauge(descProcessGroupMarkedExcluded, float64(exclusions[pclass]), string(pclass))
 	}
+
+	counts, err := cluster.GetProcessCountsWithDefaults()
```

Comment:
Can we add a single line comment here which specifies this metric as the last metric to be calculated in the function. As it includes a return statement when it encounters error. If in future, we define new metrics that are defined below this, on encountering error we will not calculate those as well. Another approach could be putting this in another function to avoid such complexities.

Reply:
I wouldn't add a comment here, if someone touches the code the person has to see that the return will stop the calculation of any metrics that follow after that. In the whole code base we don't comment on the last return value and I'm not sure how valuable this comment would be.

Moving it into a dedicated method sounds like a good idea.

Reply:
I just looked over it and moving this code part into a dedicated method is not that trivial otherwise we have to replicate the addGauge method and everything. I'll rase an issue to rework the structure later.